src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/CodeActions/Razor/SimplifyTagToSelfClosingCodeActionProvider.cs

```diff
@@ -0,0 +1,145 @@
+        if (owner is not MarkupTagHelperElementSyntax markupElementSyntax)
+        {
+            return SpecializedTasks.EmptyImmutableArray<RazorVSInternalCodeAction>();
+        }
+
+        // Check whether the code action is applicable to the element
+        if (!IsApplicableTo(context, markupElementSyntax))
+        {
+            return SpecializedTasks.EmptyImmutableArray<RazorVSInternalCodeAction>();
+        }
+
+        // Provide code action to simplify
+        var actionParams = new SimplifyTagToSelfClosingCodeActionParams
+        {
+            StartTagCloseAngleIndex = markupElementSyntax.StartTag.CloseAngle.SpanStart,
+            EndTagCloseAngleIndex = markupElementSyntax.EndTag.CloseAngle.EndPosition,
+        };
+
+        var resolutionParams = new RazorCodeActionResolutionParams()
+        {
+            TextDocument = context.Request.TextDocument,
+            Action = LanguageServerConstants.CodeActions.SimplifyTagToSelfClosingAction,
+            Language = RazorLanguageKind.Razor,
+            DelegatedDocumentUri = context.DelegatedDocumentUri,
+            Data = actionParams,
+        };
+
+        var codeAction = RazorCodeActionFactory.CreateSimplifyTagToSelfClosing(resolutionParams);
+        return Task.FromResult<ImmutableArray<RazorVSInternalCodeAction>>([codeAction]);
+    }
+
+    internal bool IsApplicableTo(RazorCodeActionContext context, MarkupTagHelperElementSyntax markupElementSyntax)
+    {
+        // Check whether the element is self-closing
+        if (markupElementSyntax is not (
+        { EndTag.CloseAngle.IsMissing: false } and
+        { StartTag.ForwardSlash: null } and
+        { StartTag.CloseAngle.IsMissing: false }
+        ))
+        {
+            return false;
+        }
+
+        // Check whether the element has any non-whitespace content
+        if (markupElementSyntax is { Body: { } body } && body.Any(static n => !n.ContainsOnlyWhitespace()))
+        {
+            return false;
+        }
+
+        // Get symbols for the markup element
+        if (!RazorComponentDefinitionHelpers.TryGetBoundTagHelpers(context.CodeDocument, markupElementSyntax.StartTag.Name.SpanStart, true, _logger, out var boundTagHelper, out _))
+        {
+            return false;
+        }
+
+        if (!boundTagHelper.IsComponentTagHelper)
+        {
+            return false;
+        }
+
+        // Check whether the Component must have children
+        foreach (var attribute in boundTagHelper.BoundAttributes)
+        {
+            // Parameter is not required
+            if (attribute is not { IsEditorRequired: true })
+            {
+                continue;
+            }
+
+            // Parameter is not a `RenderFragment` or `RenderFragment<T>`
+            if (!attribute.IsChildContentProperty() && !attribute.IsParameterizedChildContentProperty())
```

Comment:
```suggestion
            if (!attribute.IsChildContentProperty())
```

From documentation this is inclusive of both Parameterized and UnParameterized (non-generic?) 

```
    /// <summary>
    /// Gets a value that indicates whether the property is a child content property. Properties are
    /// considered child content if they have the type <c>RenderFragment</c> or <c>RenderFragment{T}</c>.
    /// </summary>
 ```